packages/plans-grid-next/src/components/features-grid/index.tsx

```diff
@@ -242,12 +242,21 @@ const FeaturesGrid = ( {
 	generatedWPComSubdomain,
 	isCustomDomainAllowedOnFreePlan,
 	gridSize,
+	isSpotlightOnCurrentPlan,
+	intent,
 }: FeaturesGridProps ) => {
 	const handleUpgradeClick = useUpgradeClickHandler( {
 		gridPlans,
 		onUpgradeClick,
 	} );
 
+	const gridPlanForSpotlight = usePlanForSpotlight( {
+		intent,
+		isSpotlightOnCurrentPlan,
+		plansForFeaturesGrid: gridPlans,
+		sitePlanSlug: currentSitePlanSlug,
+	} );
```

Comment:
This feels a little disconnected/incomplete to me. If we are passing in `gridPlans` for the `FeaturesGrid`, then why not pass `gridPlanForSpotlight` like previously? Also, looking at the code that we are replacing, it sounds like a wholly `plans-features-main` concern; so I would probably choose to move the hook under `plans-features-main` and pass the actual plan in a prop like before.
 
If we decide to retrieve the grid plans internally, then we can do so for everything and the above will make more sense.

WDYT? 🤔

Reply:
> Also, looking at the code that we are replacing, it sounds like a wholly plans-features-main concern; so I would probably choose to move the hook under plans-features-main and pass the actual plan in a prop like before.

Nope, sorry. Fine to keep the hook under `plans-grid-next`, as you have now. The only thing to keep in mind is we may/will end up exporting the spotlight plan as a separate component and rendering from `plans-features-main`. It will receive its own `gridPlan` prop like the rest of the grids - hence why it might be better to keep using the existing setup than letting the `FeaturesGrid` retrieve it like above.

Reply:
> we may/will end up exporting the spotlight plan as a separate component and rendering from plans-features-main. It will receive its own gridPlan prop like the rest of the grids - hence why it might be better to keep using the existing setup than letting the FeaturesGrid retrieve it like above.

👍  Updated to use the existing setup, with `gridPlanForSpotlight` as a `<FeaturesGrid />` prop: https://github.com/Automattic/wp-calypso/pull/87377/commits/675ac5e90facde8436c60c19c7868be8e5af2b1d